File: src/main/java/cl/app/Hour.java
package cl.app;

import java.time.LocalDateTime;
import java.time.ZoneId;
import java.util.Date;

/**
 * Created by Daniel on 01-03-2016.
 */
public enum Hour {

    MORNING("06~12"),
    MEDLEY("12~16"),
    EVENING("16~21"),
    NIGHT("21~06");

    private final String range;

    Hour(String range){
        this.range = range;
    }

    public String getRange() {
        return range;
    }

    public static Hour of(Date date){
        LocalDateTime ldt = LocalDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault());
        int hour = ldt.getHour();
        Hour elem = null;
        switch(hour){
            case 6:
            case 7:
            case 8:
            case 9:
            case 10:
            case 11:
            case 12:
                elem = Hour.MORNING;
                break;
            case 13:
            case 14:
            case 15:
            case 16:
                elem = Hour.MEDLEY;
                break;
            case 17:
            case 18:
            case 19:
            case 20:
            case 21:
                elem = Hour.EVENING;
                break;
            case 22:
            case 23:
            case 0:
            case 1:
            case 2:
            case 3:
            case 4:
            case 5:
                elem = Hour.NIGHT;
                break;
        }
        return elem;
    }

}

File: src/main/java/cl/app/type/Hour.scala
package cl.app.`type`

import java.time.{LocalDateTime, ZoneId}
import java.util.Date

/**
  * Created by Daniel on 02-03-2016.
  */
object Hour extends Enumeration{

  type Hour = Value

  val MORNING = Value("06~12")
  val MEDLEY = Value("12~16")
  val EVENING = Value("16~21")
  val NIGHT = Value("22~06")

  def of(date: Date): Hour.Value = LocalDateTime.ofInstant(date.toInstant, ZoneId.systemDefault).getHour match {
      case 6|7|8|9|10|11 => Hour.MORNING
      case 12|13|14|15|16 => Hour.MEDLEY
      case 16|17|18|19|20|21 => Hour.EVENING
      case 22|23|0|1|2|3|4|5 => Hour.NIGHT
  }

}
